tests/clock/ClockTest.cc

```diff
@@ -43,3 +44,35 @@ TEST(ClockTest, GetSimulationTimeFromClockPort)
     ASSERT_EQ(simTimeSeconds, expectedSimTimeSeconds);
     ASSERT_NEAR(simTimeNanoseconds, expectedSimTimeNanoseconds, tolerance);
 }
+
+TEST(ClockTest, SimulationStartsIfYARPClockAlreadySet)
+{
+    // This test is a regression test for
+    // https://github.com/robotology/gz-sim-yarp-plugins/issues/182, in which we check that the
+    // simulation starts without deadlocks even if YARP_CLOCK is set before launching it.
+
+    // ARRANGE
+
+    // Set YARP_CLOCK to /clock and check if test works
+    yarp::conf::environment::set_string("YARP_CLOCK", "/clock");
+
+    yarp::os::NetworkBase::setLocalMode(true);
+    // Maximum verbosity helps with debugging
+    gz::common::Console::SetVerbosity(4);
+    // Instantiate test fixture
+    auto modelPath = std::filesystem::path(CMAKE_CURRENT_SOURCE_DIR) / "model.sdf";
+    gz::sim::TestFixture fixture(modelPath.string());
+    fixture.Finalize();
+
+    const int iterations = 10;
+    const int deltaTns = 1e6; // 1ms
+
+    // ACT
+    fixture.Server()->Run(/*_blocking=*/true, iterations, /*_paused=*/false);
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
Probably we can try to reduce this? The test are already spending a lot of time, and in this specific case I do not think we need to wait for anything (the port registration is blocking as far as I know).

Reply:
As an alternative, we can check for the port existence in a for and just wait a small amount of time for each check, so that as soon as that port appears the test exits, and we need to wait for the full time only if the test fails.

Reply:
ok! I'll try to see if it works without delay, and in case it is necessary, I will implement your solution.